### PR TITLE
MGMT-2239 Fix occasional failure when cluster is updated with "nothing have changed" message

### PR DESCRIPTION
**Pull request by ori-amizur**
/cc @YuviGold 
/cc @RazRegev 
/cc @filanov 
/hold

### CODE REVIEW

File: internal/cluster/common.go [hunk 1]
```diff
@@ -68,7 +75,7 @@ func UpdateCluster(log logrus.FieldLogger, db *gorm.DB, clusterId strfmt.UUID, s
 	// Status is required as well to avoid races between different components.
 	dbReply := db.Model(&common.Cluster{}).Where("id = ? and status = ?", clusterId, srcStatus).Updates(updates)
 
-	if dbReply.Error != nil || dbReply.RowsAffected == 0 {
+	if dbReply.Error != nil || (dbReply.RowsAffected == 0 && !clusterExistsInDB(db, clusterId, updates)) {
```
Comment (filanov) [inline internal/cluster/common.go:78]: Isn't it the same as ignoring `(dbReply.RowsAffected == 0 && !clusterExistsInDB(db, clusterId, updates))` ? 
We are trying to update cluster so in 99.9% it exists in the DB.

Comment (ori-amizur) [reply internal/cluster/common.go:78]: If the cluster with the same fields that we want to update is already in the database, it is not an error.  If there is no change in the database record, then gorm will return 0 rows affected, but it is not an error.

Comment (ori-amizur) [reply internal/cluster/common.go:78]: The same already exists for hosts

Comment (filanov) [reply internal/cluster/common.go:78]: You compare only id so you don't really know if it's the same or not.

Comment (ori-amizur) [reply internal/cluster/common.go:78]: No. The updates map is also sent to the function.  So it is part if the "where".

**Review by filanov — APPROVED** [review]

**Pull request merged**